### PR TITLE
Improvements to the `SiStripHitEfficiencyHarvester` for Run 3 data-taking 

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

As per request of the SiStrip offline group, I apply few improvement for the harvesting step:
   * in case the harvester is NOT run at PCL (flag `isAtPCL` set to false) produce the final good module/ bad module `TGraphError` plot for the hit efficiency;
   *  in case the harvester is NOT run at PCL (flag `isAtPCL` set to false) produce a `TTree` to store per-DetId information for further post-processing from experts / shifters;
   * introduced a `SiStripHitEffData` struct to store the per event `FEDError`s and some events statistics;
   * profited of the PR to clean-up slightly also `SiStripHitEffFromCalibTree`: used lower case to name methods and use `   usesResource(TFileService::kSharedResource)` for threading efficiency.
   
#### PR validation:

Run the following commands:

```console
cmsDriver.py stepReAlCa -s ALCA:PromptCalibProdSiStripHitEff --conditions 123X_dataRun2_v2 --scenario pp --data --era Run2_2018 --datatier ALCARECO --eventcontent ALCARECO --processName=ReAlCa -n 100000 --dasquery='file dataset=/StreamExpress/Run2018D-SiStripCalMinBias-Express-v1/ALCARECO run=325172' --nThreads=4
```

followed by:

```console
cmsDriver.py stepHarvest -s ALCAHARVEST:SiStripHitEff --conditions 123X_dataRun2_v2 --scenario pp --data --era Run2_2018 --filein file:PromptCalibProdSiStripHitEfficiency.root -n -1
```

and obtained the following plot:

image

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport but needs to be backported for the data-taking.

### CODE REVIEW

**Comment by mmusich** [comment]
hold
    * still some work is needed before merge, but I am glad to start taking comments.

File: CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc [hunk 1]
```diff
@@ -213,38 +270,98 @@ void SiStripHitEfficiencyHarvester::dqmEndJob(DQMStore::IBooker& booker, DQMStor
...
-          badModules[det] = eff;
-          tkMapBad.fillc(det, 255, 0, 0);
-          LOGPRINT << "Layer " << layer << " (" << ::layerName(layer, showRings_, nTEClayers_) << ")  module "
-                   << det.rawId() << " efficiency: " << eff << " , " << num << "/" << denom;
-        } else {
-          //Fill the bad list with empty results for every module
-          tkMapBad.fillc(det, 255, 255, 255);
+      // use only the "good" modules
+      if (stripQuality_->getBadApvs(det) == 0) {
+        const auto eff = num / denom;
```
Comment (tvami) [inline CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc:275]: if the `auto` for the `num` and `denom` is an `uint`, will the `auto` for the `eff` somehow take care of the integer division?

Comment (mmusich) [reply CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyHarvester.cc:275]: The code was already like that before, and as far as I can tell does the right thing.

File: CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyWorker.cc [hunk 2]
```diff
@@ -947,13 +977,18 @@ void SiStripHitEfficiencyWorker::fillForTraj(const TrajectoryAtInvalidHit& tm,
         h_goodLayer.fill(layerWithSide, !badflag);
 
         // efficiency with bad modules excluded
-        if (TKlayers) {
-          h_module.fill(iidd, !badflag);
-        }
+        //if (TKlayers) {
+        //  h_module.fill(iidd, !badflag);
+        // }
```
Comment (tvami) [inline CalibTracker/SiStripHitEfficiency/plugins/SiStripHitEfficiencyWorker.cc:982]: I assume this is still the WIP part

**Comment by mmusich** [comment]
please test

**Comment by tvami** [comment]
I see this in the logs
```
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_5_X_2022-07-04-2300/src/DQMServices/Core/interface/MonitorElement.h:9,
                 from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_5_X_2022-07-04-2300/src/DQMServices/Core/interface/DQMStore.h:4,
                 from src/CalibTracker/SiStripHitEfficiency/interface/SiStripHitEffData.h:4:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_5_X_2022-07-04-2300/src/DQMServices/Core/interface/DQMNet.h:4:10: fatal error: classlib/iobase/Socket.h: No such file or directory
    4 | #include "classlib/iobase/Socket.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
unhold
   * some more work would still be desireable, but I can't follow-up quickly on this,  while the current changes are needed for operations, so I won't hold it up further.

**Comment by tvami** [comment]
+1
 * diffs seen are in the Strips ALCARECO folder, and are expected: https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_5_X_2022-07-05-1100+05d064/51404/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVDSIPIXELCALRUN1+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5+ALCAHARVD7+ALCAHARVD8/AlCaReco_SiStripHitEfficiency.html
 * PR is in line with the describtion

**Comment by tvami** [comment]
> while the current changes are needed for operations, so I won't hold it up further.

Does this deserve an `urgent` label then @mmusich ?

**Comment by mmusich** [comment]
> Does this deserve an urgent label then

I wouldn't say it's urgent, as far as I know the hit efficiency measurement is still being done with calibration trees anyway, so this will become relevant once the switchover is done in the shifter tools.

**Comment by tvami** [comment]
Ok, so not necessarily needed for 12_4_3

**Comment by perrotta** [comment]
+1

**Pull request merged**